docs/index-server.md

```diff
@@ -0,0 +1,238 @@
+In case executor(s) become dead/unavailable then the segments that were
+earlier being handled by those would be reassigned to some other
+executor using the distribution logic.
+
+**Note:** Cache loading would be done again in the new executor for the
+current query.
+
+## MetaCache DDL
+The show metacache DDL has a new column called cache location will indicate whether the cache is
+from executor or driver. To drop cache the user has to enable/disable the index server using the
+dynamic configuration to clear the cache of the desired location.
+
+## Fallback
+In case of any failure the index server would fallback to embedded mode
+which means that the JDBCServer would take care of distributed pruning.
+A similar job would be fired by the JDBCServer which would take care of
+pruning using its own executors. If for any reason the embedded mode
+also fails to prune the datamaps then the job would be passed on to
+driver.
+
+**NOTE:** In case of embedded mode a job would be fired to clear the
+cache as data cached in JDBCServer executors would be of no use.
+
+## Writing splits to a file
+If the response is too huge then it is better to write the splits to a file so that the driver can
+read this file and create the splits. This can be controlled using the property 'carbon.index.server
+.inmemory.serialization.threshold.inKB'. By default, the minimum value for this property is 0,
+meaning that no matter how small the splits are they would be written to the file. Maximum is
+102400KB which will mean if the size of the splits for a executor cross this value then they would
+be written to file.
+
+The user can set the location for these file by using 'carbon.indexserver.temp.path'. By default
+table path would be used to write the files.
+
+## Security
+The security for the index server is controlled through 'spark.carbon.indexserver.keytab' and 'spark
+.carbon.indexserver.principal'. These allow the RPC framework to login using the principal. It is
+recommended that the principal should be a super user, and the user should be exclusive for index
+server so that it does not grant access to any other service. Internally the operations would be
+executed  as a Privileged Action using the login user.
+
+The Index Server is a long running service therefore the 'spark.yarn.keytab' and 'spark.yarn
+.principal' should be configured.
+
+## Configurations
+
+##### carbon.properties(JDBCServer) 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning for the whole application.       |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+| carbon.disable.index.server.fallback | false | Whether to enable/disable fallback for index server
```

Comment:
better to give link to section explaining the fallback

Reply:
done